src/v/cluster/tm_stm.h

```diff
@@ -113,17 +124,25 @@ class tm_stm final : public persisted_stm {
         }
         return r;
     }
+    bool check_tx_membership(const kafka::transactional_id& tx_id);
 
     ss::future<checked<model::term_id, tm_stm::op_status>> barrier();
     ss::future<checked<model::term_id, tm_stm::op_status>>
       sync(model::timeout_clock::duration);
     ss::future<checked<model::term_id, tm_stm::op_status>> sync() {
         return sync(_sync_timeout);
     }
+    ss::future<tm_stm::op_status> try_init_hosted_transactions(
```

Comment:
`model::term_id` should go first like in the other methods (the whole update is conditioned on it being right so it takes precedent over the other args)